### PR TITLE
Do not scan known third party libraries

### LINKED ISSUE #744: Linter does not ignore known libs
The linter marks known third party libraries using dispensary but does not ignore them during scanning. It should though, as this is the main point about marking them known.

### PR DESCRIPTION
**Pull request by wagnerand**
Fixes #744

### CODE REVIEW

File: src/linter.js [hunk 1]
```diff
@@ -408,7 +408,14 @@ export default class Linter {
         return this.io.getFiles();
       })
       .then((files) => {
-        return this.scanFiles(Object.keys(files));
+
+        function excludeKnownLibsFilter(file) {
```
Comment (tofumatt) [inline src/linter.js]: Avoid inline functions; they will be harder to test but also are harder to follow.

Additionally, I would return a new list of files, then pass that to `scanFiles`. More lines will make things clearer, I think.

**Comment by tofumatt** [comment]
You'll need to fix the style issues in your PR: https://travis-ci.org/mozilla/addons-linter/jobs/136506550#L1435

File: tests/test.linter.js [hunk 2]
```diff
@@ -884,6 +884,47 @@ describe('Linter.extractMetadata()', function() {
...
+        return Promise.resolve(Object.keys(fakeFiles));
+      }
+    }
+
+    return addonLinter.extractMetadata({
+      _console: fakeConsole,
+      _Xpi: FakeXpi,
+    }).then((metadata) => {
+      var warnings = addonLinter.collector.warnings;
+      var filtered = warnings.map((element) => {
```
Comment (tofumatt) [inline tests/test.linter.js]: This is a good test but a comment above it, just to explain to someone a bit less familiar with the code, would be great.

File: src/linter.js (hunk 1 above)
Comment (wagnerand) [inline src/linter.js]: Ah, I did not to that bc that new list of files was only used once :)

**Comment by wagnerand** [comment]
Updated.

File: src/linter.js [hunk 3]
```diff
@@ -408,7 +408,11 @@ export default class Linter {
         return this.io.getFiles();
       })
       .then((files) => {
-        return this.scanFiles(Object.keys(files));
+        // Known libraries do not need to be scanned
+        let unknownFiles = Object.keys(files).filter((file) => {
```
Comment (tofumatt) [inline src/linter.js]: This is much better, but can we rename this to something a bit more clear? How about `filesWithoutJSLibraries` (or `filesWithoutJSLibs` if it's too long).

**Comment by tofumatt** [comment]
r+wc

**Comment by wagnerand** [comment]
Thank you!

**Pull request merged**